pulpcore/app/models/fields.py

```diff
@@ -35,7 +35,11 @@ def pre_save(self, model_instance, add):
                                'Artifact storage. Files must be stored outside this location '
                                'prior to Artifact creation.'))
         file = super().pre_save(model_instance, add)
-        if file and file._committed and add:
+        if file and file._committed and add and str(file) != upload_to:
+            # the file needs to be moved into place
             file._file = TemporaryDownloadedFile(open(file.name, 'rb'))
             file._committed = False
```

Comment:
It seems like this strangeness was already here, but we're calling super() twice with the same arguments. Should we reorder these statements so we are calling it only once?

Reply:
I tried looking at how to reorder this to only call super() once but I am not sure. This code is confusing and hard to follow. Would it make sense to open a separate refactor issue in redmine to try to rewrite this logic?

Reply:
I'm ok with an issue filed now to refactor this later.

Reply:
Ok, here we go: https://pulp.plan.io/issues/4382

Reply:
Thanks!